geomesa-native-api/src/main/java/org/locationtech/geomesa/api/GeoMesaIndex.java

```diff
@@ -0,0 +1,35 @@
+package org.locationtech.geomesa.api;
+
+import com.vividsolutions.jts.geom.Geometry;
+
+import java.util.Date;
+
+/**
+ * GeoMesaIndex is an API for utilizing GeoMesa as a spatial index
+ * without bringing in the Geotools SimpleFeature data model
+ * @param <T>
+ */
+public interface GeoMesaIndex<T> {
+
+    /**
+     * Query a GeoMesa index
+     * @param query
+     * @return
+     */
+    Iterable<T> query(GeoMesaQuery query);
+
+    /**
+     * Put a value in the GeoMesa index
+     * @param t
+     * @param geometry
+     * @param dtg
+     */
+    void put(T t, Geometry geometry, Date dtg);
```

Comment:
Yeah I guess the question is do you want an explicit feature ID type API and if so you'll want to have ```get``` ```insert``` and ```update``` methods.  I've always thought a ```put``` implies some sort of key or identifier anyways so I'd change this to ```insert``` even if we don't have an explicit id (though in English put this thing at this time and geometry sounds appealing)